lib/thesis/models/page.ex

```diff
@@ -52,7 +53,17 @@ defmodule Thesis.Page do
   def changeset(page, params \\ %{}) do
     page
     |> cast(params, @valid_attributes)
+    |> unique_constraint(:slug)
     |> validate_required(@required_attributes)
+    |> parameterize_slug
+  end
+
+  defp parameterize_slug(changeset) do
+     if {:data, slug} = fetch_field(changeset, :slug) do
```

Comment:
Can you change this to a case statement? For some reason I like that better in this case, like you did in ecto_store.